app/routes/analysis/analysis.py

```diff
@@ -48,16 +85,22 @@ async def zonal_statistics(
         "end_date": end_date
     }
 
-    response = get_lambda_client().invoke(
-        FunctionName=RASTER_ANALYSIS_LAMBDA_NAME,
-        InvocationType="RequestResponse",
-        Payload=bytes(json.dumps(payload), "utf-8"),
-    )
-
-    response_payload = json.loads(response['Payload'].read().decode())
+    response_payload_encoded = await _invoke_lambda(payload)
+    response_payload = json.loads(response_payload_encoded.decode())
 
     if response_payload['statusCode'] != 200:
         raise InvalidResponseError(f"Raster analysis returned status code {response_payload['statusCode']}")
 
     response_data = response_payload['body']['data']
     return Response(data=response_data)
+
```

Comment:
In line 92 raise `HTTPException` with status code 400 or 500 instead (user/ server error) and add a more meaningful error message. End user will see this error when making a call.